### PR TITLE
Implementation of core library with new api

### PR DESCRIPTION
**Pull request by acetechnologist**

### CODE REVIEW

File: stats/aggregation.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
+// Copyright 2017 Google Inc.
```
Comment (bogdandrutu) [inline stats/aggregation.go]: We changed the copyright
// Copyright 2017, OpenCensus Authors

File: README.md [hunk 2]
```diff
@@ -1,7 +1,153 @@
...
+    change2 := key2.CreateChange(10, TagOpInsert)           // key2 is a KeyInt64 so its value is an int64
+    builder := stats.NewTagsBuilder()
+    builder.StartFromEmpty()
+    builder.Apply(c1,c2)
+    tags := builder.Build()
+
+To create a new tag set from scratch using direct values (fastest):
+
+    builder := stats.NewTagsBuilder()
+    builder.StartFromEmpty()
```
Comment (AlekseySanin) [inline README.md]: Do we really need it? Aren't we starting empty all the time?

File: README.md [hunk 3]
```diff
@@ -1,7 +1,153 @@
...
+Create/load measures definitions:
+
+    RPCclientErrorCount   := stats.NewMeasureDescFloat64("/rpc/client/error_count", "RPC Errors", count)
+    RPCclientRequestBytes := stats.NewMeasureDescFloat64("/rpc/client/request_bytes", "Request bytes", bytes)
+    ...
+
+Register measures:
+
+	stats.RegisterMeasureDesc(RPCclientErrorCount)
+    stats.RegisterMeasureDesc(RPCclientRequestBytes)
```
Comment (AlekseySanin) [inline README.md]: spacing

File: stats/collector.go [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+	// window is the window under which the aggregation is performed.
+	w Window
+}
+
+func (c *collector) addSample(s string, v interface{}, now time.Time) {
+	aggregator, ok := c.signatures[s]
+	if !ok {
+		var newAggregationValue func() AggregationValue
+		switch agg := c.a.(type) {
```
Comment (AlekseySanin) [inline stats/collector.go]: Can this be done with inheritance vs switch?

File: README.md (hunk 2 above)
Comment (acetechnologist) [reply README.md]: No. Most of the time you need to start from an existing tagSet that you want to add to or update. We had to kill this branch so PTAL @ the newly active pull request.
The readme.md file wasn't up to date. I updated it in the new pull request.

File: README.md (hunk 3 above)
Comment (acetechnologist) [reply README.md]: Should be fixed now in the new PR

File: stats/collector.go (hunk 4 above)
Comment (acetechnologist) [reply stats/collector.go]: I think it can. Yes will fix it.

**Pull request closed without merging**